### ISSUE TITLE
Is this a floating point problem?

### ISSUE THREAD

**Issue by stacksmith** [rxi/fe#30]
A simple loop locks up:
```
(= i 0)
(while (< i 19000000)
   (= i (+ i 1)))
```
Is there a preferred way to count?

**Comment by ooichu** [comment]
Yes, this is a problem related to floating point numbers. After number 16777216 for 4-byte float the accuracy drops, and incrementation has no effect.
